evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -772,43 +786,57 @@ where
 
         // If the depth is 0, then this is the root call terminating
         if data.journaled_state.depth() == 0 {
-            for (address, expecteds) in &self.expected_calls {
-                for (expected, actual_count) in expecteds {
-                    let ExpectedCallData { calldata, gas, min_gas, value, count } = expected;
+            for (address, calldatas) in &self.expected_calls {
```

Comment:
As a note-to-self, this and all other "handlers" on the different hooks should probably be abstracted into their own function and properly documented for easy modification later.